packages/api/src/controllers/helpers.test.ts

```diff
@@ -182,6 +183,40 @@ describe("controllers/helpers", () => {
       );
     });
   });
+
+  describe("getObjectStoreS3Config", () => {
+    it("should handle a few URL formats", async () => {
+      const result = await getObjectStoreS3Config({
+        url: "s3+https://user:password@example.com/testbucket",
+      });
+      expect(result).toEqual({
+        accessKeyId: "user",
+        secretAccessKey: "password",
+        credentials: {
+          accessKeyId: "user",
+          secretAccessKey: "password",
+        },
+        region: "ignored",
+        bucket: "testbucket",
+        signingRegion: "ignored",
+        endpoint: `https://example.com/testbucket`,
+        forcePathStyle: true,
+        s3BucketEndpoint: true,
+      });
+    });
+
+    it("should reject urls with subpaths", async () => {
+      let threw = false;
+      try {
+        await getObjectStoreS3Config({
+          url: "s3+https://user:password@example.com/testbucket/subpath/foo",
+        });
+      } catch (e) {
+        threw = true;
+      }
+      expect(threw).toBe(true);
```

Comment:
😎 
```suggestion
      await expect(getObjectStoreS3Config({
          url: "s3+https://user:password@example.com/testbucket/subpath/foo",
        }).rejects.toThrow("Invalid OS URL path: /testbucket/subpath/foo");
```

Reply:
Ahhh I tried this except without the `rejects` part